File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string res="";
        int cnt=0;
        char pre='$';
        for(char c:s)
        {
            if(cnt==0)
            {
                res+=c;
                pre=c;
                cnt=1;
            }
            else
            {
                if(pre==c)
                {
                    if(cnt==1)
                    res+=c,cnt++;
                }
                else
                res+=c,cnt=1,pre=c;
            }
        }
        return res;
    }
};

File: 1302-delete-characters-to-make-fancy-string/delete-characters-to-make-fancy-string.py
class Solution:
    def makeFancyString(self, s: str) -> str:
        res=""
        cnt=0
        pre='#'
        for c in s:
            if cnt==0:
                res+=c
                cnt=1
                pre=c
            else:
                if pre==c:
                    if cnt==1:
                        res+=c
                        cnt+=1
                else:
                    res+=c
                    cnt=1
                    pre=c
        return res
        